app/controllers/posts_controller.rb

```diff
@@ -1,16 +1,47 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
-    @posts = @user.posts
+    @id = params[:user_id]
+    @user = User.find(@id)
+    @posts = User.all_post(@id)
   end
 
   def show
+    # @post = Post.find(params[:id])
```

Comment:
- [ ] To keep your code clean and professional please remove the commented code.